### PR TITLE
[ISSUE #10117]Add registration metrics.

### PR DESCRIPTION
**Pull request by stone-98**
…egistration.

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Add registration metrics.
About #10117 .

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by stone-98** [comment]
这个instance用instanceId作为指标的labels不大合适，对于批次注册实例的场景我应该通过什么来区分实例合适呢？

File: client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/NamingGrpcClientProxy.java [hunk 1]
```diff
@@ -199,10 +201,34 @@ public void doBatchRegisterService(String serviceName, String groupName, List<In
...
     }
     
+    /**
+     * Records batch registration metrics for a group of service instances.
+     *
+     * @param request   The batch instance registration request.
+     * @param exception The exception, if any, that was caught during batch registration; null if no exception
+     *                  occurred.
+     */
+    private void recordBatchRegistrationMetrics(BatchInstanceRequest request, Exception exception) {
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/NamingGrpcClientProxy.java:223]: 这样一次注册失败，会有N条metrics，不合理

**Review by KomachiSion — CHANGES_REQUESTED** [review]
感觉把简单的问题复杂化了。

其实只需要记录错误的返回码（NacosException），异常类型，和服务名即可。

**Comment by stone-98** [comment]
@KomachiSion 大佬，我这个pr的思路是以实例维度记录他的注册状态，所以对于批次注册我会记录多个，我理解你的意思是以服务名称的维度单单记录他注册失败的相关信息就好了嘛？

**Comment by stone-98** [comment]
ok，这样的话我就改成register_failures_total（注册失败总数）类型为Counter，lables为namespace、group、异常类型、异常码和服务名，我稍后调整一下。

File: client/src/main/java/com/alibaba/nacos/client/monitor/MetricsMonitor.java [hunk 2]
```diff
@@ -33,6 +33,10 @@ public class MetricsMonitor {
             .labelNames("module", "method", "url", "code").name("nacos_client_request").help("nacos_client_request")
             .register();
     
+    private static final Gauge NACOS_REGISTER_STATUS_MONITOR = Gauge.build()
+            .name("nacos_client_register_status_monitor").help("nacos_client_register_status_monitor")
+            .labelNames("service_name", "address", "protocol", "register_type").register();
```
[the remark this answers is not in the document]
Comment (stone-98) [reply client/src/main/java/com/alibaba/nacos/client/monitor/MetricsMonitor.java]: 已调整

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**